### DISCUSSION TITLE
How should we structure pattern documents?

### DISCUSSION THREAD

**Discussion by roberthstrand** [open-gitops/project#11] [Q&A]
As we discussed in this weeks prinicples meeting, we should probably think a little bit about how we structure the so-called "patterns" documents that we all want done.

What I would suggest, based on the points brought up in the meeting, is that we create a pattern directory in the documents repository. The way I see it, we could either have one directory per pattern, so that we don't clutter that one directory with a lot of images, or we could have every document serialized in that one directory with it's respective images.

Any thoughts about this? I think I prefer to have one directory per pattern.

We want to have some metadata for possible later usage, where one could add tools and plattforms. How should we structure this? Have one for platform, one for CI, one for CD?

**Comment by christianh814** [comment]
For a lack of a better idea...what about this?

```
├── patterns
│   ├── thing1
│   │   ├── docs
│   │   └── images
│   ├── thing2
│   │   ├── docs
│   │   └── images
│   └── thing3
│       ├── docs
│       └── images
├── reference-implementations
│   ├── thing1
│   │   ├── docs
│   │   └── images
│   ├── thing2
│   │   ├── docs
│   │   └── images
│   └── thing3
│       ├── docs
│       └── images
└── usecases
    ├── thing1
    │   ├── docs
    │   └── images
    ├── thing2
    │   ├── docs
    │   └── images
    └── thing3
        ├── docs
        └── images
```

**Comment by christianh814** [comment]
I guess it also will depend where these documents will "live". I don't think we got a consensus during the meeting. Personally, I think these should go with the "docs". But not against them going in the "website". The only caveat with the "website" is we would have to follow whatever HUGO wants (which isn't a big deal, but something to think about). cc @scottrigby @todaywasawesome
